openstf/pipeline/train_model.py

```diff
@@ -135,15 +135,18 @@ def train_model_pipeline_core(
 
     # Split data
     train_data, validation_data, test_data = split_data_train_validation_test(
-        data_with_features.sort_index(axis=1)
+        data_with_features.sort_index(axis=0)
     )
 
     # Create relevant model
     model = ModelCreator.create_model(pj)
 
     # split x and y data
-    train_x, train_y = train_data.iloc[:, 1:], train_data.iloc[:, 0]
-    validation_x, validation_y = validation_data.iloc[:, 1:], validation_data.iloc[:, 0]
+    train_x, train_y = train_data.iloc[:, 1:-1], train_data.iloc[:, 0]
+    validation_x, validation_y = (
+        validation_data.iloc[:, 1:-1],
+        validation_data.iloc[:, 0],
+    )
```

Comment:
Here also, perhaps wise to add a check on column names to make sure the last column should be excluded